packages/@expo/cli/src/utils/remote-build-cache-providers/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import { type Options as AndroidRunOptions } from '../../run/android/resolveOptions';
+import { type Options as IosRunOptions } from '../../run/ios/XcodeBuild.types';
+import { hasDirectDevClientDependency } from '../../start/detectDevClient';
+
+export function isDevClientBuild({
+  runOptions,
+  projectRoot,
+}: {
+  runOptions: AndroidRunOptions | IosRunOptions;
+  projectRoot: string;
+}) {
+  if (!hasDirectDevClientDependency(projectRoot)) {
+    return false;
+  }
+
+  if ('variant' in runOptions && runOptions.variant !== undefined) {
+    return runOptions.variant === 'debug';
+  }
+  if ('configuration' in runOptions && runOptions.configuration !== undefined) {
+    return runOptions.configuration === 'Debug';
+  }
```

Comment:
```suggestion
  if ('variant' in runOptions) {
    return runOptions.variant !== 'debug';
  }
  if ('configuration' in runOptions) {
    return runOptions.configuration !== 'Debug';
  }
```

nit: comment for the downstack pr.

Reply:
`runOptions.variant !== 'debug'` would mean that this is true when `undefined` or `release`, which is not exactly what we want given that this function should return true only when it is a devClient build.

If we want to remove the `!== undefined` check, we would need to do 

```suggestion
  if ('variant' in runOptions) {
    return runOptions.variant !== 'release';
  }
  if ('configuration' in runOptions) {
    return runOptions.configuration !== 'Release';
  }
```

Reply:
that's right! i was confused from the logic. maybe i should go to bed now. thanks again for correcting me. please forget this comment 😅

Reply:
But this wouldn't cover custom build variants on Android

Reply:
No worries, thanks for review @Kudo!